src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -921,13 +944,13 @@ public static unsafe object CheckCast(MethodTable* pTargetType, object obj)
                 return CheckCastArray(pTargetType, obj);
             else if (pTargetType->IsInterface)
                 return CheckCastInterface(pTargetType, obj);
-            else if (pTargetType->IsParameterizedType)
-                return CheckCastNonArrayParameterizedType(pTargetType, obj);
+            else if (pTargetType->IsParameterizedType || pTargetType->IsFunctionPointerType)
+                return CheckCastUnboxableType(pTargetType, obj);
             else
                 return CheckCastClass(pTargetType, obj);
         }
 
-        private static unsafe object CheckCastNonArrayParameterizedType(MethodTable* pTargetType, object obj)
+        private static unsafe object CheckCastUnboxableType(MethodTable* pTargetType, object obj)
```

Comment:
```suggestion
`CheckCastUnboxableType` name does look right. UnboxableType = type that can be unboxed. Where is the type that can be unboxed in this?
```

Reply:
Would `Nonboxable` be better? I'm going for "a type that cannot be present on the heap".

Reply:
Yep

Reply:
`NonObjectType`? Things like pointers can be on the heap as an element/field of something else (unlike byrefs for example). Pointers do not have an Object form though.

Reply:
I was referring to describing "a type that cannot be present on the heap". 
`Nonboxable` seems fine in this context, since we are trying to unbox.